src/main/scala/org/broadinstitute/dsde/agora/server/ga4gh/Ga4ghService.scala

```diff
@@ -1,69 +1,70 @@
-            complete(StatusCodes.OK, Metadata())
-          }
+          complete(StatusCodes.OK, Metadata())
         } ~
         path("tool-classes") {
           val toolClassesResponse:Seq[ToolClass] = Seq(ToolClass.apply(Some(AgoraEntityType.Workflow)))
           complete(toolClassesResponse)
         } ~
-        path("tools") { requestContext =>
+        path("tools") {
           // TODO: query params and response headers
-          val message = QueryPublicTools(requestContext)
-          perRequest(requestContext, queryHandler, message)
+          complete(queryPublicTools())
         } ~
-        path("tools" / Segment) { id => requestContext =>
-          val entity = entityFromArguments(id)
-          val message = QueryPublicTool(requestContext, entity)
-          perRequest(requestContext, queryHandler, message)
+        path("tools" / Segment) { id =>
+          complete(queryPublicTool(entityFromArguments(id)))
         } ~
-        path("tools" / Segment / "versions") { id => requestContext =>
-          val entity = entityFromArguments(id)
-          val message = QueryPublic(requestContext, entity)
-          perRequest(requestContext, queryHandler, message)
+        path("tools" / Segment / "versions") { id =>
+          complete(queryPublic(entityFromArguments(id)))
         } ~
-        path("tools" / Segment / "versions" / Segment) { (id, versionId) => requestContext =>
-          val entity = entityFromArguments(id, versionId)
-          val message = QueryPublicSingle(requestContext, entity)
-          perRequest(requestContext, queryHandler, message)
+        path("tools" / Segment / "versions" / Segment) { (id, versionId) =>
+          complete(queryPublicSingle(entityFromArguments(id, versionId)))
         } ~
         path("tools" / Segment / "versions" / Segment / "dockerfile") { (id, versionId) =>
-          complete(spray.http.StatusCodes.NotImplemented)
+          complete(StatusCodes.NotImplemented)
         } ~
-        path("tools" / Segment / "versions" / Segment / Segment / "descriptor") { (id, versionId, descriptorType) => requestContext =>
-          val entity = entityFromArguments(id, versionId)
-          val message = QueryPublicSinglePayload(requestContext, entity, parseDescriptorType(descriptorType))
-          perRequest(requestContext, queryHandler, message)
+        path("tools" / Segment / "versions" / Segment / Segment / "descriptor") { (id, versionId, descriptorType) =>
+          val agoraEntity: Future[AgoraEntity] = queryPublicSingleEntity(entityFromArguments(id, versionId))
+          val descriptor = parseDescriptorType(descriptorType)
+          onComplete(agoraEntity) {
+            case Success(ae) =>
+              descriptor match {
+                case ToolDescriptorType.WDL => complete(ToolDescriptor(ae))
+                case ToolDescriptorType.PLAIN_WDL =>
+                  val payload: String = ae.payload.getOrElse("")
+                  complete(payload)
```

Comment:
In theory, this can be simplified. Doing so, however, gives a `ToResponseMarshallable` error - it thinks the string is an `Object` and doesn't compile.